source.go

```diff
@@ -83,20 +107,38 @@ func (s *MultipleSource) Close() error {
 
 func (s *MultipleSource) createSubscribeHandler(prefixSize int, dist *distribute) nats.MsgHandler {
 	return func(msg *nats.Msg) {
+		// call the callback for on subscription received. this will allow users to plugin metrics if they desire.
+		s.opts.onSubscriptionReceived(msg)
```

Comment:
Given `s.opts.onSubscriptionReceived` is not initialised by default, this would result in `nil` here unless the `SourceOptOnSubscriptionReceived` functional option is provided. 

I would recommend we the construction of `sourceOpt` defining a default noop implementation, OR check for nil upon usage (as has been done for `s.opts.onMessageRelayed` below).

Reply:
added wrapper funcs to `sourceOptions` to handle nil case